aws/api.ts

```diff
@@ -67,7 +67,7 @@ export class API implements cloud.API {
     private readonly staticRoutes: StaticRoute[];
     private readonly proxyRoutes: ProxyRoute[];
     private readonly routes: Route[];
-    private readonly customDomains: Domain[];
+    private readonly customDomains: cloud.Domain[];
```

Comment:
I believe it was actually intentional to use `Domain` and not `cloud.Domain` here so that additional properties available on the AWS-specific side would be visible to callers using `cloudAws.API`.  In general, I think this is a pattern we'll need to do more of where the provider-specific implementations expose derived interfaces with additional details for consumers willing to admit they are using that provider.

Reply:
Interesting... i'll revert this then.  I'm not sure i'm a fan of htat approach as it makes the code exceptionally leaky, and can make it so that people can write against 'cloud' but end up locking themselves into a specific provider.

Reply:
That's sort of the whole point of `cloud-aws` - it's a richer implementation of a narrower interface.  You use the richer AWS specific pieces when you are okay opting-in to platform-specific functionality.  In practice, this is more or less the only way to provide graceful ability to mix-n-match AWS specific things with `cloud`.  If you have thoughts on alternate approaches to the two specific places this capability is used here though - definitely interested in discussing.